pkg/ingester/client/pool.go

```diff
@@ -10,33 +10,32 @@ import (
 	"github.com/weaveworks/common/user"
 	"github.com/weaveworks/cortex/pkg/util"
 	context "golang.org/x/net/context"
-	grpc_health_v1 "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/health/grpc_health_v1"
 )
 
 // Factory defines the signature for an ingester client factory
-type Factory func(addr string, cfg Config) (IngesterClient, error)
+type Factory func(addr string) (grpc_health_v1.HealthClient, error)
```

Comment:
I think pushing this type into the public interface makes it clear the coupling between `IngesterPool` and gRPC is unhealthy.
How about defining an abstract interface containing some `IsHealthy()` method and push the gRPC details back to (in this case) IngesterClient ?